pkg/monitord/host.go

```diff
@@ -0,0 +1,124 @@
+package monitord
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/network"
+	"github.com/threefoldtech/zos/pkg/upgrade"
+	"github.com/vishvananda/netlink"
+)
+
+// HostMonitor monitor host information
+type hostMonitor struct {
+	duration time.Duration
+
+	// version related
+	watcher *fsnotify.Watcher
+	version semver.Version
+	boot    upgrade.Boot
+}
+
+// NewHostMonitor initialize a new host watcher
+func NewHostMonitor(duration time.Duration) (pkg.HostMonitor, error) {
+	if duration == 0 {
+		duration = 2 * time.Second
+	}
+	watcher, err := fsnotify.NewWatcher()
```

Comment:
Do you still needs this watcher here ? Since the version is not pushed from identityd itself when a new version is available.

Reply:
yup, this need to go idd.